runtime/runtime/src/prefetch.rs

```diff
@@ -304,6 +328,73 @@ impl TriePrefetcher {
         Ok(())
     }
 
+    /// Prefetcher tuned for claim.sweat::claim contract calls.
+    ///
+    /// Remove after #10965 reaches mainnet.
+    fn prefetch_claim_sweat_claim(
+        &self,
+        account_id: AccountId,
+        predecessor: AccountId,
+    ) -> Result<(), PrefetchError> {
+        let Self { prefetch_api, trie_root, .. } = self;
+        let trie_root = *trie_root;
+        let prefetch_api = prefetch_api.clone();
+        rayon::spawn(move || {
+            let mut account_data_key = Vec::with_capacity(4 + 8 + predecessor.len());
+            let Ok(()) = 0u8.serialize(&mut account_data_key) else { return };
+            let Ok(()) = predecessor.serialize(&mut account_data_key) else { return };
+            let trie_key =
+                TrieKey::ContractData { account_id: account_id.clone(), key: account_data_key };
+            // Just read this directly for now since this is temporary anyway
+            let prefetcher_storage = prefetch_api.make_storage();
+            let trie = Trie::new(prefetcher_storage, trie_root, None);
+            let Ok(Some(account_record)) = trie.get(&trie_key.to_vec()) else {
```

Comment:
The `trie` is based on `TriePrefetchingStorage`, which [should ensure](https://github.com/near/nearcore/blob/0efb5f71d138503f64625676d6b46cdfdcaa4970/core/store/src/trie/prefetching_trie_storage.rs#L218) the data is not retrieved simultaneously by multiple threads